app/controllers/two_factor_authentication/options_controller.rb

```diff
@@ -28,16 +46,19 @@ def two_factor_options_presenter
     end
 
     def process_valid_form
-      factor_to_url = {
-        'voice' =>  otp_send_url(otp_delivery_selection_form: { otp_delivery_preference: 'voice' }),
-        'personal_key' => login_two_factor_personal_key_url,
-        'sms' => otp_send_url(otp_delivery_selection_form: { otp_delivery_preference: 'sms' }),
-        'auth_app' => login_two_factor_authenticator_url,
-        'piv_cac' => login_two_factor_piv_cac_url,
-        'webauthn' => login_two_factor_webauthn_url,
-      }
-      url = factor_to_url[@two_factor_options_form.selection]
-      redirect_to url if url
+      url = mfa_redirect_url
+      redirect_to url if url.present?
+    end
+
+    def mfa_redirect_url
+      selection = @two_factor_options_form.selection
+      options = EXTRA_URL_OPTIONS[selection] || {}
+
+      configuration_id = @two_factor_options_form.configuration_id
+      options[:id] = configuration_id if configuration_id.present?
+
+      method = FACTOR_TO_URL_METHOD[selection]
+      send(method, options) if method.present?
```

Comment:
Could we use `public_send` please? https://github.com/rubocop-hq/ruby-style-guide#prefer-public-send

Reply:
Definitely. That was at the back of my mind, but I didn't take the time to track it down.